modules/@apostrophecms/piece-type/index.js

```diff
@@ -1206,6 +1206,43 @@ module.exports = {
 
           console.log(`Done unlocalizing module ${self.name}`);
         }
+      },
+
+      touch: {
+        usage: 'Invoke this task to touch (update without any change) all docs of this type.',
+        async task(argv) {
+          const req = self.apos.task.getAdminReq();
+          let errCount = 0;
+          let count = 0;
+
+          // We have 30 minutes (by default) for each iterration.
+          // https://www.mongodb.com/docs/manual/reference/method/cursor.noCursorTimeout/#session-idle-timeout-overrides-nocursortimeout
+          const cursor = (await self.find(req, {})
+            .locale(null)
+            .limit(0)
+            .sort(false)
+            .toMongo())
+            .addCursorFlag('noCursorTimeout', true);
+
+          for await (const doc of cursor) {
+            try {
+              await self.update(req, doc);
+              count++;
+            } catch (e) {
+              errCount++;
+              self.apos.util.error(e);
+            }
+          }
+
+          console.log(`Touched ${count} doc(s) with ${errCount} error(s)`);
+
+          // Return, useful for tests and internal API's
+          // It's in effect only when invoked via apos.task.invoke().
```

Comment:
Right, another reason to make sure we clean up the cursor if we somehow bomb.

Reply:
:100: